dgl_sparse/include/sparse/sparse_format.h

```diff
@@ -90,6 +95,21 @@ std::shared_ptr<CSR> COOToCSC(const std::shared_ptr<COO>& coo);
 /** @brief Convert a CSR format to CSC format. */
 std::shared_ptr<CSR> CSRToCSC(const std::shared_ptr<CSR>& csr);
 
+/** @brief Convert a Diag format to COO format. */
+std::shared_ptr<COO> DiagToCOO(
```

Comment:
Why do we need this 3 conversion? In which case, they will be used?

Reply:
They are used for operators that do not have implementation on Diag format, e.g., SpMM.

Reply:
What's the implication of the performance here if we convert diag to COO/CSR/CSV for operators?
Will it strongly decrease the spmm performance?

This solution looks good to me, but let's make sure we understand the trade-off we are making here.

Reply:
I simply follow our current implementation to ensure no performance regression. Currently, we also convert the DiagMatrix to SparseMatrix for SpMM on the Python side.